src/plugins/orangehrmOpenidAuthenticationPlugin/Controller/OpenIdConnectRedirectController.php

```diff
@@ -18,36 +18,160 @@
+use OrangeHRM\Core\Traits\Auth\AuthUserTrait;
+use OrangeHRM\Core\Traits\ORM\EntityManagerHelperTrait;
+use OrangeHRM\Entity\User;
 use OrangeHRM\Framework\Http\RedirectResponse;
 use OrangeHRM\Framework\Http\Request;
+use OrangeHRM\Framework\Routing\UrlGenerator;
+use OrangeHRM\Framework\Services;
+use OrangeHRM\I18N\Traits\Service\I18NHelperTrait;
+use OrangeHRM\OpenidAuthentication\Traits\Service\SocialMediaAuthenticationServiceTrait;
 
 class OpenIdConnectRedirectController extends AbstractVueController implements PublicControllerInterface
 {
+    use AuthUserTrait;
+    use I18NHelperTrait;
+    use SessionHandlingTrait;
+    use EntityManagerHelperTrait;
+    use SocialMediaAuthenticationServiceTrait;
+
     /**
      * @var HomePageService|null
      */
     protected ?HomePageService $homePageService = null;
 
+    /**
+     * @var null|LoginService
+     */
+    protected ?LoginService $loginService = null;
+
     /**
      * @return HomePageService
      */
     public function getHomePageService(): HomePageService
     {
-        if (!$this->homePageService instanceof HomePageService) {
-            $this->homePageService = new HomePageService();
-        }
-        return $this->homePageService;
+        return $this->homePageService ??= new HomePageService();
     }
 
-    //TODO
+    /**
+     * @return LoginService
+     */
+    public function getLoginService(): LoginService
+    {
+        return $this->loginService ??= new LoginService();
+    }
+
+    /**
+     * @throws Exception
+     */
     public function handle(Request $request): RedirectResponse
     {
-        $response = $this->getResponse();
+        $this->beginTransaction();
+        /** @var UrlGenerator $urlGenerator */
+        $urlGenerator = $this->getContainer()->get(Services::URL_GENERATOR);
+        $loginUrl = $urlGenerator->generate('auth_login', [], UrlGenerator::ABSOLUTE_URL);
+
+        try {
+            $providerId = $this->getAuthUser()->getAttribute(AuthUser::PROVIDER_ID);
+            $authProvider = $this->getSocialMediaAuthenticationService()->getAuthProviderDao()
+                ->getAuthProviderById($providerId);
+            $authProviderExtraDetails = $this->getSocialMediaAuthenticationService()->getAuthProviderDao()
+                ->getAuthProviderDetailsByProviderId($providerId);
 
-        $homePagePath = $this->getHomePageService()->getHomePagePath();
-        return $this->redirect($homePagePath);
+            $oidcClient = $this->getSocialMediaAuthenticationService()->initiateAuthentication(
+                $authProviderExtraDetails,
+                $this->getSocialMediaAuthenticationService()->getScope(),
+                $this->getSocialMediaAuthenticationService()->getRedirectURL()
+            );
+
+            $oidcClient->authenticate();
+
+            $userCredentials = new UserCredential();
+            $userCredentials->setUsername($oidcClient->requestUserInfo('email'));
```

Comment:
see whether we need to handle error if `$oidcClient->requestUserInfo('email')` this is null